src/components/Common/BaseContextMenu.tsx

```diff
@@ -0,0 +1,175 @@
+import { IconDotsVertical } from '@tabler/icons-react';
+import { Fragment, useMemo } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getByHighlightColor } from '@/src/utils/app/folders';
+
+import {
+  BaseContextMenuProps,
+  BaseMenuItemRendererProps,
+} from '@/src/types/menu';
+
+import { Menu, MenuItem } from './DropdownMenu';
+import { Tooltip, TooltipContent, TooltipTrigger } from './Tooltip';
+
+function BaseContextMenuItemRenderer({
+  name,
+  Icon,
+  dataQa,
+  onClick,
+  disabled,
+  highlightColor,
+  translation,
+  className,
+  menuItems,
+}: BaseMenuItemRendererProps) {
+  const { t } = useTranslation(translation);
```

Comment:
I'm not sure that passing it as a param is good idea, because you can have different cases where 1 namespace is not enough
SO I think it's better to set final translation on place where you declaring values for context menu